src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -331,9 +364,41 @@ public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, Create
             if (flags.HasFlag(CreateObjectFlags.Aggregation))
                 throw new NotImplementedException();
 
-            object? wrapperMaybeLocal = wrapperMaybe;
-            retValue = null;
-            throw new NotImplementedException();
+            if (!flags.HasFlag(CreateObjectFlags.UniqueInstance))
+            {
+                var comInterfaceDispatch = impl.TryGetComInterfaceDispatch(externalComObject);
```

Comment:
This should be done only for `CreateObjectFlags.Unwrap`.

Reply:
Question. Should roundrtip from Marshal API happens automatically? For me this is means that I should create global `ComWrappers` with `CreateObjectFlags.Unwrap` instead of `CreateObjectFlags.None`.
```
var target = new ComObject();
int result = CaptureComPointer(target);
ThrowIfNotEquals(0, result, "Seems to be COM marshalling behave strange.");
result = RetreiveCapturedComPointer(out var capturedObject);
ThrowIfNotEquals(0, result, "Seems to be COM marshalling behave strange.");
if (capturedObject != target)
{
    throw new Exception("Should round-trip");
}
```

Reply:
Probably. I do not remember the details like this. Test what CoreCLR does and match it.

Reply:
Will do. It was unexpected twist that you do not know details. From outside appears that you do know details.

Reply:
This works as I expect. Add that to smoke tests.